### DISCUSSION TITLE
How to use next js image optimization when images are served from another microservice in kubernetes?

### DISCUSSION THREAD

**Discussion by HsnVahedi** [vercel/next.js#22639] [Help]
I have a react component in my `nextjs` application which uses `next/image`:

    <Image
        src={`http://backend${image.url}`}
        alt={image.alt}
        width={image.width}
        height={image.height}
        />

`backend` is the name of a kubernetes serivce through which images are available to `nextjs` microservice.

When I deploy it, `nextjs` microservice crashes (with no errors) when rendering the `<Image />` components. (Pages with no `<Image />` get rendered without any problem.)

I don't know the reason but the `pod` restarts the `nexjs` container. I checked `kubectl top pod` to see whether it's because of resource limitations but found no CPU/RAM limitations.

The images are located in a provisioned kubernetes volume. There are two `nginx` microservices which serve the images:

 1. One of them serves the images to world, for example web browsers can request images like this: `/media/images/image-1.jpg` (which works fine.)
 2. The other one serves the images to the `nextjs` microservice. It includes a kubernetes service named `backend`. For example in `nextjs` container an image can be requested like this:

    `<Image src={'http://backend${image.url}' ... />`

When I turn off image optimization (`unoptimized={true}`) and change the `src` to get the image from the web browser (instead of `nextjs` microservice), everything works fine:

    <Image
        unoptimized={true}
        src={`${image.url}`}
        alt={image.alt}
        width={image.width}
        height={image.height}
        />

I want the image optimization feature. How to handle this problem. Thanks in advance!

Is there any example which has used `next/image` in a `kubernetes` cluster?

**Comment by kruseio** [comment]
Have you included your backend domain into the image config whitelist?

<br>

next.config.js
```
module.exports = {
  images: {
    domains: ['backend'],
  },
}
```

**Reply by HsnVahedi** [reply]
Hello @kruserr ! Yes I did.

Actually the problem is solved now. check [this](https://github.com/vercel/next.js/issues/22644) out. Thanks for your response.
